### ISSUE TITLE
Allow selective build cache debug logging

### ISSUE THREAD

**Issue by w25r** [gradle/gradle#2241]
### Expected Behavior
Allow users to specify which types of logs they're interested in finer logging. This can be done perhaps by task, by category, or something else.

### Current Behavior
Build Cache logs meant specifically for debugging cache hits show up at the info level:
```
...
Appending inputPropertyHash for 'reports.enabledReportNames' to build cache key: 6920849d8fe8162599fd903f04036bf1
Appending inputPropertyHash for 'reports.html.class' to build cache key: c521c47cb0f6005948938a07487ec2b7
Appending inputPropertyHash for 'reports.html.displayName' to build cache key: 87c363ec67df19154ea51636f7572ed0
Appending inputPropertyHash for 'reports.html.enabled' to build cache key: f6d7ed39fe24031e22d54f3fe65b901c
Appending inputPropertyHash for 'reports.html.name' to build cache key: 45b63c04ebea9f3558fe60946eb8d98b
Appending inputPropertyHash for 'reports.html.outputType' to build cache key: 0c5a7f9ee0b1630048617c6a027bb3ce
Appending inputPropertyHash for 'reports.junitXml.class' to build cache key: c4652ec214f00ac83f067720baa20acd
Appending inputPropertyHash for 'reports.junitXml.displayName' to build cache key: 527b168d48ec1929dfcb1c6a7c220137
Appending inputPropertyHash for 'reports.junitXml.enabled' to build cache key: f6d7ed39fe24031e22d54f3fe65b901c
Appending inputPropertyHash for 'reports.junitXml.name' to build cache key: 1fca07d69580ccd153cbc5aff057ccf5
Appending inputPropertyHash for 'reports.junitXml.outputPerTestCase' to build cache key: c06857e9ea338f3f3a24bb78f8fbdf6f
Appending inputPropertyHash for 'reports.junitXml.outputType' to build cache key: 0c5a7f9ee0b1630048617c6a027bb3ce
Appending inputPropertyHash for 'scanForTestClasses' to build cache key: f6d7ed39fe24031e22d54f3fe65b901c
Appending inputPropertyHash for 'systemProperties' to build cache key: cfcc5f871f42c7c797557fd17748dc17
Appending inputPropertyHash for 'bootstrapClasspath' to build cache key: d41d8cd98f00b204e9800998ecf8427e
Appending inputPropertyHash for 'candidateClassFiles' to build cache key: 291f4a48d7c09cf3914c2028e9edf8b7
Appending inputPropertyHash for 'classpath' to build cache key: 8002332050d86e1509d21a07afa742c6
Appending inputPropertyHash for 'nonEmptyCandidateClassFiles' to build cache key: 291f4a48d7c09cf3914c2028e9edf8b7
...
```

### Context
The debate has already been had to some degree with @bigdaz and @wolfs on this [commit](https://github.com/gradle/gradle/commit/dce2ff794a5ac18a7bfc09f5067982fad4f2cf8e#commitcomment-22318528):

We must be careful not to impose our logging decisions on users. 

We used to have logging level defined per task, but the solution was slow.

**Comment by lptr** [comment]
I agree with the general sentiment. However, we currently need a way to show this data to users who don't yet have access to Gradle Enterprise.  For 4.0 we reached the compromise with @alkemist and @etiennestuder to bring this data back from `DEBUG` to `INFO` level. For 4.1 we should be able to find a better solution.

**Comment by eriwen** [comment]
@lptr any changes here for 4.1, or should this be deferred?

**Comment by eriwen** [comment]
This may be related to #1010

**Comment by ilya-g** [comment]
I can confirm that reading INFO logs after this change became much harder.

Have you considered enabling inputPropertyHash logging only when some system property is set?

**Comment by h0tk3y** [comment]
This is still an issue for us. Reading the `--info` logs with build cache enabled seems to be harder than it should.

**Comment by lptr** [comment]
We are considering doing this for Gradle 4.6.

**Comment by xenoterracide** [comment]
hope so, or perhaps some way of disabling them optionally, e.g. `--disable-log=classname`, also my CI server, seems to (for some reason) have log length limit, so I'm not sure if I'll even be able to read them all, but without --info the logging is next to useless

**Comment by wolfs** [comment]
This has been fixed by #4206.

We only log the input hashes when the Gradle property `org.gradle.caching.debug` is set to `true`.
